common/expression/src/type_check.rs

```diff
@@ -322,15 +330,30 @@ pub fn common_super_type(ty1: DataType, ty2: DataType) -> Option<DataType> {
         (DataType::Array(box ty1), DataType::Array(box ty2)) => {
             Some(DataType::Array(Box::new(common_super_type(ty1, ty2)?)))
         }
-        (DataType::UInt8, DataType::UInt16) | (DataType::UInt16, DataType::UInt8) => {
```

Comment:
oooops, seems that the case for U8 and I8 was missing.

Reply:
what's the behavior of `(u8,i8)` 🤔，I think this pattern should return None

Reply:
I think it should be i16. An exception is (u64, i64), and in this case we can return i64.

Reply:
There is  [type_coercion.rs](https://github.com/datafuselabs/databend/blob/8c3730f25e5beaff68c3eee1f2228f01a0faad42/common/datavalues/src/types/type_coercion.rs)